plutus-core/plutus-core/src/PlutusCore/Builtin/HasConstant.hs

```diff
@@ -24,33 +23,24 @@ throwNotAConstant
     => Maybe cause -> m r
 throwNotAConstant = throwingWithCause _UnliftingError "Not a constant"
 
-class AsConstant term where
+-- | Ensures that @term@ has a 'Constant'-like constructor to lift values to and unlift values from.
+class HasConstant term where
     -- Switching from 'MonadError' to 'Either' here gave us a speedup of 2-4%.
     -- | Unlift from the 'Constant' constructor throwing an 'UnliftingError' if the provided @term@
     -- is not a 'Constant'.
     asConstant
         :: AsUnliftingError err
         => Maybe cause -> term -> Either (ErrorWithCause err cause) (Some (ValueOf (UniOf term)))
 
-class FromConstant term where
     -- | Wrap a Haskell value as a @term@.
     fromConstant :: Some (ValueOf (UniOf term)) -> term
```

Comment:
nit: maybe `asConstant` should be `toConstant` for symmetry.

Reply:
I use the convention of calling things that can fail `as*` and things that can't `to*/from*`. We have that with errors too, `AsParseError`, `AsTypeError` etc.

Reply:
Ah I see